node/pkg/lifecycle/startup/startup.go

```diff
@@ -215,18 +216,84 @@ func Run() {
 		log.WithError(err).Errorf("Unable to ensure network for os")
 		utils.Terminate()
 	}
+}
+
+// ManageNodeCondition updates the Kubernetes node condition on successful startup and then sleeps forever.
+func ManageNodeCondition() {
+	ctx := context.TODO()
+
+	if os.Getenv("CALICO_NETWORKING_BACKEND") == "none" {
+		// Calico is not managing networking, so we don't need to set the NetworkUnavailable condition.
+		log.Info("Calico is not managing networking, skipping NetworkUnavailable condition update")
+		<-ctx.Done()
+		return
+	}
+
+	k8sNodeName := utils.DetermineNodeName()
+	if nodeRef := os.Getenv("CALICO_K8S_NODE_REF"); nodeRef != "" {
+		k8sNodeName = nodeRef
+	}
+
+	config, err := winutils.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	if err != nil {
+		log.WithError(err).Error("Failed to build Kubernetes config")
+		utils.Terminate()
+	}
+	config.Timeout = 2 * time.Second
+
+	// Create the k8s clientset.
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		log.WithError(err).Error("Failed to create clientset")
+		utils.Terminate()
+	}
+
+	// Determine which components should be checked for readiness. We don't do any liveness checking here.
+	// Do this by checking the contents of /etc/service/enabled.
+	checkBIRD := false
+	if _, err := os.Stat("/etc/service/enabled/bird/run"); err == nil {
+		checkBIRD = true
+	}
+	checkBIRD6 := false
+	if _, err := os.Stat("/etc/service/enabled/bird6/run"); err == nil {
+		checkBIRD6 = true
+	}
+
+	// We always check felix.
+	checkFelix := true
```

Comment:
This needs to be configured based on whether or not Felix has its health check enabled.